### PR TITLE
add: Add dev dependencies to user setup.cfg

### PR DESCRIPTION
**Pull request by joshlynchONS**
# Summary

Add an optional dev dependencies section to the users setup file. This helps the user manage the different dependencies between their package users and developers

# Checklists

This pull/merge request meets the following requirements:

- [ ] code runs
- [ ] [developments are ethical][data-ethics-framework] and secure
- [ ] you have made proportionate checks that the code works correctly
- [ ] you have tested the build of the template
- [ ] test suite passes
- [ ] [minimum usable documentation][agilemodeling] written in the `docs` folder

Comments have been added below around the incomplete checks.

[agilemodeling]: http://agilemodeling.com/essays/documentLate.htm
[data-ethics-framework]: https://www.gov.uk/government/publications/data-ethics-framework

### CODE REVIEW

File: {{ cookiecutter.repo_name }}/README.md [hunk 1]
```diff
@@ -23,6 +23,14 @@ using:
 pip install -e .
```
Comment (Jacobb164) [inline {{ cookiecutter.repo_name }}/README.md:23]: Do we need the pip install here or should it be make install?

Also there is a bit at the bottom of the README.md which say about doing pip install requirements. Can this be replaced / removed and can you check in the other docs it's not there

File: {{ cookiecutter.repo_name }}/docs/contributor_guide/CONTRIBUTING.md [hunk 2]
```diff
@@ -10,18 +10,17 @@ we'd be happy to help!
...
+To start contributing, open your terminal, and install the package and it's required
+Python dependencies using:
 
 ```shell
-pip install -r requirements.txt
-pre-commit install
+pip install -e .[dev]
 ```
 
-or the `make` command:
```
[the remark this answers is not in the document]
Comment (Jacobb164) [reply {{ cookiecutter.repo_name }}/docs/contributor_guide/CONTRIBUTING.md:21]: This still needs to be updated

File: {{ cookiecutter.repo_name }}/README.md (hunk 1 above)
Comment (joshlynchONS) [reply {{ cookiecutter.repo_name }}/README.md:24]: Replaced the requirements at the bottom to use the makefile. I've left pip install as an option as well as the Makefile. Do you think I should just keep the Makefile install?

File: {{ cookiecutter.repo_name }}/docs/contributor_guide/CONTRIBUTING.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (joshlynchONS) [reply {{ cookiecutter.repo_name }}/docs/contributor_guide/CONTRIBUTING.md:21]: This has been changed to offer the make command and the pip install

**Pull request merged**